NumLib/DOF/DOFTableUtil.cpp

```diff
@@ -30,4 +44,60 @@ std::vector<GlobalIndexType> getIndices(
     return indices;
 }
 
+double norm(GlobalVector const& x, unsigned const global_component,
+            MathLib::VecNormType norm_type,
+            LocalToGlobalIndexMap const& dof_table, MeshLib::Mesh const& mesh)
+{
+    // TODO that also includes ghost nodes.
+    std::size_t const n_nodes = mesh.getNumberOfNodes();
+
+    // TODO for PETSc some global accumulation is necessary.
```

Comment:
For the norms of whole vectors there are already functions in `MathLib/LinAlg/LinAlg.h`, which use the PETSc routines in the respective case.
The norm function introduced here, however, only picks, e.g., every third element of the vector and computes a norm involving only those elements. Therefore I think this function should not be moved to the vector classes.
And I didn't check if PETSc supports such "masked" norm computations out of the box.